### ISSUE TITLE
Site Settings: "Unsaved changes..." alert displays incorrectly

### ISSUE THREAD

**Issue by roccotripaldi** [Automattic/wp-calypso#11682]
To reproduce:

1. Select a site, and navigate to the General Settings page.
2. Update your Related Post settings, and click Save.
3. Get a green success message.
4. Try to navigate away from page, and get an "Unsaved changes..." alert

**Comment by jeremeylduvall** [comment]
I experienced this the other day with the site privacy setting. I kept trying to flip a JP site from Hidden to Public. I would save the changes, but I got the same "Unsaved changes..." alert every time I tried to leave the page. In my case though, the alert was displaying correctly meaning if I exited anyway and refreshed my settings, the site was still on Hidden.

I experienced this with Related Posts as well. I turned them on, saved, tried to exit, received the alert, and exited anyway. When I refreshed, the Related Posts option was still turned off so my changes hadn't actually been saved.
